### DISCUSSION TITLE
ContractFunctionExecutionError: Execution reverted for an unknown reason.

### DISCUSSION THREAD

**Discussion by ombayoeco11** [wevm/wagmi#2494] [General]
Why is this happening?
Raw Call Arguments:
  from:  0x70b55C90ff8bd602FfB8dFB8dBBBc2e7B6b1D943
  to:    0xA5F03693280d1752D92093e7b7CCFD15560873E7
  data:  0x61b7a2120000000000000000000000000000000000000000000000000000000000000003
 
Contract Call:
  address:   0xA5F03693280d1752D92093e7b7CCFD15560873E7
  function:  BuyItem(uint32 _productid)
  args:             (3)
  sender:    0x70b55C90ff8bd602FfB8dFB8dBBBc2e7B6b1D943

Docs: https://viem.sh/docs/contract/simulateContract.html
Details: execution reverted
Version: viem@0.3.37
    at getContractError (getContractError.ts:52:1)
    at simulateContract (simulateContract.ts:150:1)
    at async prepareWriteContract (chunk-OKULXQAF.js:2078:1)

and here's the code i've been using
const BuyItems = async () => {
    if (isProgress) return;
    setIsProgress(true); 
      try {
        const { request } = await prepareWriteContract({
          address: NFTADDR,
          abi: NFTABI,
          functionName: "BuyItem",
          args: [id],
        });
       
          const { hash } = await writeContract(request);
          Swal.fire({
            title: "Please Wait",
            html: "Waiting for the transaction to be finished...",
            timerProgressBar: true,
            didOpen: () => {
              Swal.showLoading();
            },
          });
          const data = await waitForTransaction({ hash });

          if (data) {
            Swal.fire({
              icon: "success",
              title: "Horray..",
              text: "Transaction" + hash,
            });
          }
        
      } catch (error) {
        console.log(error);
        Swal.fire({
          icon: "error",
          title: "Oops",
          text: "Transaction Failed " + error,
        });
      } finally {
        setIsProgress(false);
      }
    
  };

    return (
    <div className="btn-wrapper flex items-start w-full">
        <button
          onClick={BuyItems}
          disabled={isProgress}
          className="bg-sky-500 px-6 py-3 text-[#14233d] font-['Orbitron'] hover:transition-all hover:duration-300 hover:bg-[#14233d] hover:text-white"
        >
          Buy NFT
        </button>
        <p className="text-white">{allow}</p>
      </div>
      )

i'm using CRA node 16
is there anybody can help me?

**Comment by jxom** [comment]
It means that either the contract function reverted, or you do not have enough funds in your account to fund the transaction.

**Reply by ombayoeco11** [reply]
that's only one of the problems, when i call "allowance" it also gave me errors. i forgot to screenshot the errors. 
But perhaps you are right...maybe i dont' have enough fund
